src/app/app.ts

```diff
@@ -107,11 +121,40 @@ export const appActions = {
       runtime: {
         ...state.runtime,
         showEditor: true,
+        showPreview: false,
         selectedNode: nodeId,
       },
     }
   },
 
+  selectNodeHover: ({nodeId, pos}: {nodeId: GraphNodeId, pos: IPosition}) =>
+    (state: IState) => {
+      return {
+        runtime: {
+          ...state.runtime,
```

Comment:
This, `...state.runtime` isn't actually needed it turns out. The framework does a shallow merge already. It's only needed if you're making a deep update (i.e. top level props on an object will be diff'ed already and merge with the previous version of the object)